### DISCUSSION TITLE
Основополагающие функции

### DISCUSSION THREAD

**Discussion by Better-hack** [OnlyFart/Elib2Ebook#10] [Ideas]
1) устанавливаемый тайм аут между запросами. 
(по типу --Time 150 --Rand 20   , будет означать что запрос раз в 150+-20 секунд)
2) запросы через прокси.
(в виде --ProxyIP '29.207.12.83' --ProxyPort 20 --ProxyPass 'f4df3g5'  , где если не указан пароль, то значит его нету)
3) возможность установки начальной и конечной главы. К примеру из 1700 глав нужны  только с 200-ой по 700-ую.
( --ChapterStart 200 --ChapterEnd 700)

**Comment by OnlyFart** [comment]
Пункт 2 в некотором виде уже реализован. В данный момент только для http прокси и без пароля

**Reply by Better-hack** [reply]
Что по первому и третьему пункту?

**Reply by OnlyFart** [reply]
По первому пункту мне непонятно зачем это делать. Никаких проблем связанных с таймаутом между запросами мною замечено не было

По третьему пункту готово #25

**Reply by Better-hack** [reply]
> По первому пункту мне непонятно зачем это делать. Никаких проблем связанных с таймаутом между запросами мною замечено не было
> 
Правильная настройка при помощи первого пункта позволяет скачивать, не получая бана от сайта. (Я один такой уже получил)
Я как бы обычно не сильно спешу скачать все главы, а вот прога подряд запросы отправляет без остановки, это всё видно в админ панели как "странный трафик", поэтому много где при помощи рандомного тайм аута и указания максимального количества запросов в xx секунд. справляются с данной проблемой. 
Вот причины бана с некоторых ресурсов: "Подозрение на скачивание глав, для другого сайта", "бот атака", "Не соблюдение пользовательского соглашения";

**Reply by Better-hack** [reply]
>  ![image](https://user-images.githubusercontent.com/66402475/185955025-6e8bc584-ee45-4059-8132-44333aa684dd.png)

https://wuxiaworld.ru/

**Reply by OnlyFart** [reply]
С этим сайтом ситуация сложная. Мои недолгие эксперименты никак не привели в успешным попыткам избежать бана. Единственный вариант с указанием очень большого интервала >= 5 секунд.

**Comment by VasyK-1989** [comment]
Добрый день
Возможно вопрос не в тему, но может кто подсказать как правильно пользоваться программой. При запуске Elib2Ebook-Windows-portable появляется черное окно, максимум на 1-1,5 секунды, ничего больше сделать не успеваю. Это я программу не правильно скачал или нужно что-то еще жать?
